### PR TITLE
Split Madagascar into 6 provinces

### PR DESCRIPTION
**Pull request by strump**
For new outer Madagascar border I picked relation [#447325](https://www.openstreetmap.org/relation/447325). Internal borders are defined from `admin_level = 3` relations [#3203752](https://www.openstreetmap.org/relation/3203752), [#3247585](https://www.openstreetmap.org/relation/3247585), [#3203754](https://www.openstreetmap.org/relation/3203754), [#3203753](https://www.openstreetmap.org/relation/3203753), [#3203751](https://www.openstreetmap.org/relation/3203751), [#3203749](https://www.openstreetmap.org/relation/3203749)
See [new partitioning map](http://umap.openstreetmap.fr/en/map/madagascar-pie_843093#6/-19.064/48.764).

Removed Madagascar.poly and added 6 new:

| Part name | File size |
| --------------- | --------------- |
| `Madagascar_Antananarivo province` | 58 MB |
| `Madagascar_Antsiranana province` | 41 MB |
| `Madagascar_Fianarantsoa province` | 123 MB |
| `Madagascar_Mahajanga province` | 11 MB |
| `Madagascar_Toamasina province` | 15 MB |
| `Madagascar_Toliara province` | 23 MB |

Updated translations using Wikidata.

### Uploaded to Android for testing:

| Capital city | Madagascar folder |
| --------------- | --------------- |
| screen_madagascar_capital  | screen_madagascar_downloads |

### Issues:
<s>1. Android app still askes to download full Madagascar map:</s>
<s>2. Search by province name doesn't show the province but full Madagascar instead:</s>

### Questions
1. <s>Could those issues be caused by online countries catalog, which still has Madagascar as one map? Or this is because outdated `packed_polygons.bin` file? Let me update packed borders and check.</s> After updating `packed_polygons.bin` issues gone.
2. I use 6 provinces of Madagascar. But [wiki says](https://en.wikipedia.org/wiki/Provinces_of_Madagascar) that from 2007 those provinces are not used anymore. Madagascar is divided into 23 regions (faritras) now. Would it be OK for locals to see old provinces in OM maps list?
3. Should I also update `data/borders_vs_osm.csv`, `data/old_vs_new.csv`, `data/countries_meta.txt`, `data/mwm_names_en.txt` ?

### CODE REVIEW

File: tools/python/maps_generator/utils/file.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ def is_verified(name: AnyStr) -> bool:
 def copy_overwrite(from_path: AnyStr, to_path: AnyStr):
     if os.path.exists(to_path):
         shutil.rmtree(to_path)
-    shutil.copytree(from_path, to_path)
```
Comment (strump) [inline tools/python/maps_generator/utils/file.py:192]: if `from_path` points to a file (not a directory), then `shutil.copytree` fails with exception. Replaced with `shutil.copy`

**Comment by strump** [comment]
One more thing. [Node density map](https://tyrasd.github.io/osm-node-density/#7/-20.938/47.153/2021,places) shows disproportion of data in [Fitovinany region](https://osm.org/relation/6951285).

Screenshot 2022-12-06 at 13 21 05

A lot of `landuse=farmland` objects. Looks like [Operation Fistula](https://wiki.openstreetmap.org/wiki/Organised_Editing/Activities/Operation_Fistula) is responsible for those objects.

**Comment by vng** [comment]
This is totally great! A good start to push big work like splitting MWMs.

I will make a separate planet build with your changes. We _should_ test flows like:
- updating maps
- routing when we have old and new data downloaded

**Comment by vng** [comment]
Did you make any useful scripts to fetch borders from OSM? Good to have them also in repo. Or make a Doc on how to split a big country, regenerate polygons, etc.

Our current bottleneck is Vietnam, Kenya, Tanzania

**Comment by vng** [comment]
https://github.com/organicmaps/organicmaps/pull/1277

**Comment by strump** [comment]
@vng  I has a Python script to export border from OSM relation to POLY/GPX format https://github.com/strump/osm-borders-to-poly. But in my case it requires additional editing after export. OM expects POLY with no holes. But multipoligons in OSM could have islands. So I used Python script, and then run JOSM to compile OSM data into POLY files.

**Comment by biodranik** [comment]
1. @vng should we add holes support? (Vatican, San Marino, something else)
2. @strump It would be great to have scripts that simplify borders support.
3. New polygons should be mentioned in countries.txt. Does it also have a field to store the "old big" region?
4. Using outdated regions is a bad idea. Can we predict that in 3-5 years OSM data density will increase, and splitting to current 23 regions for example will be ok in the size? @vng how big will be the overhead for 23 vs 7 regions?
5. Routing errors can be forcedly fixed by constantly showing the same alert dialog "Some maps are outdated, routing may not work properly, please update all maps on the route" with buttons "Update" and "Use old data".
6. Do we really have some working solution now to properly split the big map into smaller ones, without breaking user's experience? Or it should be coded?

**Comment by biodranik** [comment]
You made a good point there too. For islands like this one, the best option is to use a large polygon with as few points as possible, because we don't need detailed coastline borders in our algorithms. More points slow down the processing.

File: tools/python/maps_generator/var/etc/file_generation_order.txt [hunk 2]
```diff
@@ -909,7 +909,12 @@ Russia_Primorsky Krai	1091.039607
 Russia_Bryansk Oblast	1089.024201
 Italy_Lombardy_Como	1087.4186009999999
 Ukraine_Volyn Oblast	1084.5126600000003
-Madagascar	1067.3086999999998
+Madagascar_Antananarivo province
```
Comment (biodranik) [inline tools/python/maps_generator/var/etc/file_generation_order.txt:912]: What are these numbers that are missing?

**Review by biodranik — COMMENTED** [review]
The main issue now with splitting any region is "how can we smoothly upgrade existing users with existing downloaded regions"?

I assume that this problem was already solved previously, by introducing an "old" field in data/countries.txt. @strump can you please investigate and test if it works?

@vng WDYT?

Another important issue to keep in mind, that all .poly files _should share existing borders_ with each region's neighbor, to avoid "holes".

**Comment by newsch** [comment]
Relates to #2283

**Pull request closed without merging**